pkgs/_macro_client/lib/src/execute_macro.dart

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:dart_model/dart_model.dart';
+import 'package:macro/macro.dart';
+import 'package:macro_service/macro_service.dart';
+
+/// Runs [macro] in the types phase and returns a  [AugmentResponse].
+Future<AugmentResponse> executeTypesMacro(
+    Macro macro, Host host, AugmentRequest request) async {
+  final target = request.target;
+  final queryResult = await host.query(Query(target: target));
```

Comment:
This is okay for now but will need follow-up, I guess a TODO and (if you like) a GitHub issue? Same TODO on lines 44 and 74.

The question is what the host should initially send to the macro and how it gets passed to the macro code. For example if we decide the macro always wants this query result then it can already be sent in `AugmentRequest`, no need for a round trip.

Reply:
Yeah long term I think we want just the declaration but not members of the declaration.